Quick replies

feychou opened the pull request:
https://github.com/lemmings-io/02-facebook-example/pull/17#issuecomment-311128764

feychou committed:
Add matchers and quick-reply

feychou committed:
Change default matcher

feychou committed:
Make happy

feychou committed:
Cleanup

feychou committed:
Change route back to /webhook (#18)

feychou committed:
Add warning

feychou committed:
Use quick-reply template

feychou committed:
MOve on-attachment call before on-message and remove sender-id from comments

feychou committed:
Document callbacks

feychou committed:
Update bot.clj

sido378 reviewed (approved):
looks good imo.
the only thing left to discuss would be the formatting of payloads (uppercase vs kebab-case)
I'm pretty indifferent, but I think it should be uniform across all payloads to make the difference between payload and text content more obvious. 
That argument would also speak for uppercase, so maybe just change `get-help` and `get-started` to uppercase?

feychou commented:
@sido378 Yeah, I agree. It seems to be the standard in Messenger API documentation and I think it helps people not to confuse the text with the payload when they are passed as arguments.

On a different note, I would love to be able to pass named arguments like

`(templates/quick-reply :text "Clojure" :payload "CLOJURE")`

and this is what I would mostly go for when the arguments succession becomes hard to remember or is completely arbitrary. But this is out of context, especially since we are using an external lib for templating. I will just drop [this link](https://stackoverflow.com/questions/3337888/clojure-named-arguments#answer-3337967) here.

feychou committed:
Change get started button payload to facebook conventions

https://developers.facebook.com/docs/messenger-platform/messenger-profile/get-started-button
